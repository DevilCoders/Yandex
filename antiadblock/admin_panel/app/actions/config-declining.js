export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

