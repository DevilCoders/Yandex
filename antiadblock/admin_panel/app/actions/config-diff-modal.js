export const OPEN_CONFIG_DIFF_MODAL = 'OPEN_CONFIG_DIFF_MODAL';
export const CLOSE_CONFIG_DIFF_MODAL = 'CLOSE_CONFIG_DIFF_MODAL';

export function openConfigDiffModal(serviceId, id, secondId) {
    return {
        type: OPEN_CONFIG_DIFF_MODAL,
        serviceId,
        id,
        secondId
    };
}

export function closeConfigDiffModal() {
    return {
        type: CLOSE_CONFIG_DIFF_MODAL
    };
}
