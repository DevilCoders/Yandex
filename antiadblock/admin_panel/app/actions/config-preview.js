export const OPEN_CONFIG_PREVIEW = 'OPEN_CONFIG_PREVIEW';
export const CLOSE_CONFIG_PREVIEW = 'CLOSE_CONFIG_PREVIEW';

export function openConfigPreview(serviceId, id, configData) {
    return {
        type: OPEN_CONFIG_PREVIEW,
        serviceId,
        id,
        configData
    };
}

export function closeConfigPreview(id) {
    return {
        type: CLOSE_CONFIG_PREVIEW,
        id
    };
}
