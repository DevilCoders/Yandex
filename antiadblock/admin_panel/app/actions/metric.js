import serviceApi from 'app/api/service';

export const START_METRIC_LOADING = 'START_METRIC_LOADING';
export const END_METRIC_LOADING = 'END_METRIC_LOADING';

export function fetchMetric(serviceId, name, range) {
    return dispatch => {
        dispatch(startMetricLoading(serviceId, name, range));

        return serviceApi.fetchServiceMetric(serviceId, name, range).then(data => {
            return dispatch(endMetricLoading(serviceId, name, range, data, true));
        }, () => {
            return dispatch(endMetricLoading(serviceId, name, range, []));
        });
    };
}

export function startMetricLoading(serviceId, name, range) {
    return {
        type: START_METRIC_LOADING,
        serviceId,
        name,
        range
    };
}

export function endMetricLoading(serviceId, name, range, data, loaded) {
    return {
        type: END_METRIC_LOADING,
        serviceId,
        name,
        range,
        data,
        loaded
    };
}
