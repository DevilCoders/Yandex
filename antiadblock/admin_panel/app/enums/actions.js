export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
