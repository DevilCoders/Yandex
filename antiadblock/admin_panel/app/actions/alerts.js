import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
