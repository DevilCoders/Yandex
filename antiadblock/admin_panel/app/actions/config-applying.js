import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
