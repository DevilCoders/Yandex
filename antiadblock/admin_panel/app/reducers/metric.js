import * as serviceActions from 'app/actions/service';
import * as metricActions from 'app/actions/metric';
import update from 'app/lib/update';

import {METRICS} from 'app/enums/graphs';

export default function(state, action) {
    const {type} = action;

    state = state || {};

    switch (type) {
        // При загрузке сервиса инициализируем state мониторинга пустыми объектами
        case serviceActions.END_SERVICE_LOADING: {
            return update(state, {
                $merge: {
                    [action.service.id]: {
                        ...Object.values(METRICS).reduce((acc, item) => {
                            acc[item] = {
                                loading: false,
                                loaded: false
                            };

                            return acc;
                        }, {})
                    }
                }
            });
        }
        case metricActions.START_METRIC_LOADING: {
            return update(state, {
                $merge: {
                    [action.serviceId]: {
                        ...state[action.serviceId],
                        [action.name]: {
                            range: action.range,
                            loaded: false,
                            loading: true
                        }
                    }
                }
            });
        }
        case metricActions.END_METRIC_LOADING: {
            if (state[action.serviceId][action.name].range === action.range) {
                return update(state, {
                    $merge: {
                        [action.serviceId]: {
                            ...state[action.serviceId],
                            [action.name]: {
                                data: action.data,
                                loading: false,
                                loaded: action.loaded
                            }
                        }
                    }
                });
            }

            return state;
        }
        default: {
            return state;
        }
    }
}

