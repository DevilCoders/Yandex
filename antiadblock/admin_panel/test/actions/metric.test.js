import * as metricActions from 'app/actions/metric';
import * as metricActionsFixtures from '../fixtures/actions/metric';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {METRIC_HTTP_CODES} from '../fixtures/backend/metric';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('metric actions', () => {
    const serviceId = 'test.service';
    const metricName = 'http_codes';
    const range = 336;
    const loaded = true;
    const emptyStore = {};

    it('should create START_METRIC_LOADING', () => {
        expect(metricActions.startMetricLoading(serviceId, metricName, range)).toEqual(metricActionsFixtures.START_METRIC_LOADING);
    });

    it('should create END_METRIC_LOADING', () => {
        expect(metricActions.endMetricLoading(serviceId, metricName, range, METRIC_HTTP_CODES, loaded)).toEqual(metricActionsFixtures.END_METRIC_LOADING);
    });

    it('should create START_METRIC_LOADING after loading', () => {
        const store = mockStore(emptyStore);

        fetch.mockResponseOnce(JSON.stringify(METRIC_HTTP_CODES));

        return store.dispatch(metricActions.fetchMetric(serviceId, metricName, range)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create START_METRIC_LOADING after loading and return false to loaded flag', () => {
        const store = mockStore(emptyStore);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(metricActions.fetchMetric(serviceId, metricName, range)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
