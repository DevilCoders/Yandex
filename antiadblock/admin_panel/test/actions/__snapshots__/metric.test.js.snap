// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metric actions should create START_METRIC_LOADING after loading 1`] = `
Array [
  Object {
    "name": "http_codes",
    "range": 336,
    "serviceId": "test.service",
    "type": "START_METRIC_LOADING",
  },
  Object {
    "data": Object {
      "1524758640000": Object {
        "200": 426,
        "204": 8,
        "301": 40,
        "302": 3,
        "404": 40,
      },
      "1524758700000": Object {
        "200": 2566,
        "204": 75,
        "301": 239,
        "302": 4,
        "304": 4,
        "404": 227,
      },
      "1524758760000": Object {
        "200": 2365,
        "204": 66,
        "301": 270,
        "302": 7,
        "304": 3,
        "404": 247,
      },
      "1524758820000": Object {
        "200": 2624,
        "204": 80,
        "301": 281,
        "302": 12,
        "304": 20,
        "404": 240,
        "502": 1,
      },
      "1524758880000": Object {
        "200": 3368,
        "204": 109,
        "301": 330,
        "302": 13,
        "304": 4,
        "404": 293,
        "502": 1,
      },
      "1524758940000": Object {
        "200": 2359,
        "204": 75,
        "301": 225,
        "302": 13,
        "404": 193,
      },
      "1524759000000": Object {
        "200": 2454,
        "204": 93,
        "301": 197,
        "302": 15,
        "404": 170,
      },
      "1524759060000": Object {
        "200": 2805,
        "204": 92,
        "301": 265,
        "302": 13,
        "304": 2,
        "404": 233,
      },
      "1524759120000": Object {
        "200": 2806,
        "204": 87,
        "301": 230,
        "302": 12,
        "304": 6,
        "404": 213,
      },
      "1524759180000": Object {
        "200": 2368,
        "204": 85,
        "301": 182,
        "302": 11,
        "304": 1,
        "404": 173,
      },
      "1524759240000": Object {
        "200": 2797,
        "204": 100,
        "301": 232,
        "302": 20,
        "304": 7,
        "404": 206,
      },
      "1524759300000": Object {
        "200": 2421,
        "204": 80,
        "301": 214,
        "302": 14,
        "304": 3,
        "404": 197,
      },
      "1524759360000": Object {
        "200": 2311,
        "204": 80,
        "301": 194,
        "302": 13,
        "304": 6,
        "404": 183,
      },
      "1524759420000": Object {
        "200": 2471,
        "204": 86,
        "301": 277,
        "302": 11,
        "304": 3,
        "404": 264,
      },
      "1524759480000": Object {
        "200": 3478,
        "204": 119,
        "301": 378,
        "302": 22,
        "304": 3,
        "404": 355,
      },
      "1524759540000": Object {
        "200": 2321,
        "204": 77,
        "301": 212,
        "302": 11,
        "404": 198,
      },
    },
    "loaded": true,
    "name": "http_codes",
    "range": 336,
    "serviceId": "test.service",
    "type": "END_METRIC_LOADING",
  },
]
`;

exports[`metric actions should create START_METRIC_LOADING after loading and return false to loaded flag 1`] = `
Array [
  Object {
    "name": "http_codes",
    "range": 336,
    "serviceId": "test.service",
    "type": "START_METRIC_LOADING",
  },
  Object {
    "data": Array [],
    "loaded": undefined,
    "name": "http_codes",
    "range": 336,
    "serviceId": "test.service",
    "type": "END_METRIC_LOADING",
  },
]
`;
